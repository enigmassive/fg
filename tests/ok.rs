use std::fs;

#[test]
fn test() {
    fs::read_dir("./tests/ok/").unwrap().for_each(|entry| {
        let entry = entry.unwrap();
        let name = entry.file_name().into_string().unwrap();
        let path = entry.path();

        let src = String::from_utf8(fs::read(path).unwrap()).unwrap();
        let mut dest = vec![];
        fg::compile(&src, &mut dest);
        let code = String::from_utf8(dest).unwrap();
        insta::assert_snapshot!(name, code);
    })
}
