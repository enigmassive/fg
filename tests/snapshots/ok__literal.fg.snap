---
source: tests/ok.rs
expression: code
---
package literal

func helloWorld() (string) {
	return "hello, world"
}

func theAnswer() (int) {
	return 42
}
